pkg/securitypolicy/framework.rego

```diff
@@ -223,8 +229,32 @@ enforcement_point_info := {"available": false, "allowed": false, "unknown": fals
     semver.compare(data.api.svn, enforcement_point.introducedVersion) < 0
 }
 
+default exec_external := {"allowed": false}
+
+exec_external := {"allowed": true} {
+    some process in data.policy.external_processes
+    command_ok(process.command)
+    envList_ok(process.env_rules)
```

Comment:
Once we get to GCS, the process as environment variables and we need to enforce those. I think given the API we should be doing this now. There's a set of environment variables that are coming from the host per your https://github.com/microsoft/hcsshim/blob/main/internal/cmd/cmd.go#L122 link.

Because they come from the host and arrive via the API and aren't set in the guest, we need to have enforcement and we should be doing what we are with it right now. Having it in policy and enforcing based on that. 

Anything that arrives from the host and is used by the guest for starting up a process should always be in policy and checked. Doesn't matter what we are doing on the host side at the moment, it matters what could be done in the hands of an untrusted party (which is everyone host side).

Reply:
I am not saying that we shouldn't enforce environment variables. Under normal circumstances the ONLY environment variable present will be the one I linked above. I'm not saying we shouldn't be enforcing, the question is, whether this should be in the policy. in my opinion it's not and we just define the expected environment for external process in rego.

Reply:
I've updated the PR such that a single environment variable constraint for `PATH` is added in `ExternalProcessConfig.toInternal()`. As discussed, I think it is a good compromise position in this circumstance.

Reply:
discussed offline. for general case it's beneficial to have environment variables in the policy for `exec_external`. Although it's currently not possible to do via shimdiag.